### PR TITLE
feat: implementation of getting the pokemon details

### PR DESCRIPTION
**Pull request by henryjrtena**
```
- created a freezed model for details, stats, move and ability 
- created a function for handling the api request
- added additional todos for comment to be remove in main.dart
```

image
image

### CODE REVIEW

File: lib/api/model/pokemon_ability.dart [hunk 1]
```diff
@@ -0,0 +1,25 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'pokemon_ability.freezed.dart';
+part 'pokemon_ability.g.dart';
+
+@freezed
+class PokemonAbility with _$PokemonAbility {
```
Comment (shawn-ffufm) [inline lib/api/model/pokemon_ability.dart:7]: no need to add the prefix Pokemon here, Ability is enough 🙂 

apply same thing for other data classes except for `pokemon_details.dart` just leave it as is 🙂

File: lib/api/model/pokemon_ability.dart [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  const factory PokemonAbility({
+    @JsonKey(name: 'ability') required AbilityInfo ability,
+  }) = _PokemonAbility;
+
+  factory PokemonAbility.fromJson(Map<String, dynamic> json) =>
+      _$PokemonAbilityFromJson(json);
```
Comment (shawn-ffufm) [inline lib/api/model/pokemon_ability.dart:13]: do you need this whitespace here?

File: lib/api/model/pokemon_ability.dart [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+@freezed
+class AbilityInfo with _$AbilityInfo {
```
Comment (shawn-ffufm) [inline lib/api/model/pokemon_ability.dart:17]: kindly move this to a separate dart file 🙂

File: lib/api/model/pokemon_ability.dart [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  const factory AbilityInfo({
+    @JsonKey(name: 'name') required String name,
+    @JsonKey(name: 'url') required String url,
+  }) = _AbilityInfo;
+
+  factory AbilityInfo.fromJson(Map<String, dynamic> json) =>
+      _$AbilityInfoFromJson(json);
```
Comment (shawn-ffufm) [inline lib/api/model/pokemon_ability.dart:24]: same here

File: lib/api/model/pokemon_move.dart [hunk 5]
```diff
@@ -0,0 +1,25 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'pokemon_move.freezed.dart';
+part 'pokemon_move.g.dart';
+
+@freezed
+class PokemonMove with _$PokemonMove {
```
Comment (shawn-ffufm) [inline lib/api/model/pokemon_move.dart:7]: see related feedback above for this 🙂

File: lib/api/model/pokemon_move.dart [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  const factory PokemonMove({
+    @JsonKey(name: 'move') required MoveInfo move,
+  }) = _PokemonMove;
+
+  factory PokemonMove.fromJson(Map<String, dynamic> json) =>
+      _$PokemonMoveFromJson(json);
```
Comment (shawn-ffufm) [inline lib/api/model/pokemon_move.dart:13]: see related feedback above for this 🙂

File: lib/api/model/pokemon_move.dart [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+@freezed
+class MoveInfo with _$MoveInfo {
```
Comment (shawn-ffufm) [inline lib/api/model/pokemon_move.dart:17]: see related feedback above for this 🙂

File: lib/api/model/pokemon_move.dart [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+  const factory MoveInfo({
+    @JsonKey(name: 'name') required String name,
+    @JsonKey(name: 'url') required String url,
+  }) = _MoveInfo;
+
+  factory MoveInfo.fromJson(Map<String, dynamic> json) =>
+      _$MoveInfoFromJson(json);
```
Comment (shawn-ffufm) [inline lib/api/model/pokemon_move.dart:24]: see related feedback above for this 🙂

File: lib/api/model/pokemon_stats.dart [hunk 9]
```diff
@@ -0,0 +1,27 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'pokemon_stats.freezed.dart';
+part 'pokemon_stats.g.dart';
+
+@freezed
+class PokemonStats with _$PokemonStats {
```
Comment (shawn-ffufm) [inline lib/api/model/pokemon_stats.dart:7]: see related feedback above for this 🙂

File: lib/api/model/pokemon_stats.dart [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+  const factory PokemonStats({
+    @JsonKey(name: 'base_stat') required double baseStat,
+    @JsonKey(name: 'effort') required double effort,
+    @JsonKey(name: 'stat') required StatInfo stat,
+  }) = _PokemonStats;
+
+  factory PokemonStats.fromJson(Map<String, dynamic> json) =>
+      _$PokemonStatsFromJson(json);
```
Comment (shawn-ffufm) [inline lib/api/model/pokemon_stats.dart:15]: see related feedback above for this 🙂

File: lib/api/model/pokemon_stats.dart [hunk 11]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+@freezed
+class StatInfo with _$StatInfo {
```
Comment (shawn-ffufm) [inline lib/api/model/pokemon_stats.dart:19]: see related feedback above for this 🙂

File: lib/api/model/pokemon_stats.dart [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+  const factory StatInfo({
+    @JsonKey(name: 'name') required String name,
+    @JsonKey(name: 'url') required String url,
+  }) = _StatInfo;
+
+  factory StatInfo.fromJson(Map<String, dynamic> json) =>
+      _$StatInfoFromJson(json);
```
Comment (shawn-ffufm) [inline lib/api/model/pokemon_stats.dart:26]: see related feedback above for this 🙂

**Review by shawn-ffufm — CHANGES_REQUESTED** [review]
@henryjrtena see pr feedback 🙂

**Comment by henryjrtena** [comment]
App still runs after the changes in the code.

image
image

File: lib/api/model/ability.dart [hunk 13]
```diff
@@ -0,0 +1,14 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:pokedex_async_redux/api/model/ability_info.dart';
+
+part 'ability.freezed.dart';
+part 'ability.g.dart';
+
+@freezed
+class Ability with _$Ability {
+  const factory Ability({
```
Comment (shawn-ffufm) [inline lib/api/model/ability.dart:9]: kindly add a new line on top of this to make it more readable 🙂

File: lib/api/model/ability_info.dart [hunk 14]
```diff
@@ -0,0 +1,14 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'ability_info.freezed.dart';
+part 'ability_info.g.dart';
+
+@freezed
+class AbilityInfo with _$AbilityInfo {
+  const factory AbilityInfo({
```
Comment (shawn-ffufm) [inline lib/api/model/ability_info.dart:8]: see related feedback above 🙂

File: lib/api/model/ability.dart [hunk 15]
```diff
@@ -0,0 +1,14 @@
...
+    @JsonKey(name: 'ability') required AbilityInfo ability,
+  }) = _Ability;
+  factory Ability.fromJson(Map<String, dynamic> json) =>
```
Comment (shawn-ffufm) [inline lib/api/model/ability.dart:12]: same here

**Review by shawn-ffufm — CHANGES_REQUESTED** [review]
see minor pr feedback 🙂

**Review by shawn-ffufm — APPROVED** [review]
lgtm 💪

**Pull request merged**